lib/vpi/VpiImpl.cpp

```diff
@@ -168,7 +168,7 @@ GpiObjHdl* VpiImpl::create_gpi_obj_from_handle(vpiHandle new_hdl,
 GpiObjHdl* VpiImpl::native_check_create(std::string &name, GpiObjHdl *parent)
 {
     vpiHandle new_hdl;
-    std::string fq_name = parent->get_name() + "." + name;
+    std::string fq_name = parent->get_fullname() + "." + name;
```

Comment:
the fullname is set using the name passed appended the parents name when it was created. So starting from the root if an object is found via either implementation they would have the same name. 

I do possibly want to do

```
    vpiHandle new_hdl;
    std::string search_name = parent->get_name() + "." + name;
    std::vector<char> writable(seach_name.begin(), search_name.end());
    writable.push_back('\0');

    new_hdl = vpi_handle_by_name(&writable[0], NULL);
    if (new_hdl == NULL) {
        LOG_DEBUG("Unable to query vpi_get_handle_by_name %s", name.c_str());
        return NULL;
    }
    std::string fq_name = parent->get_fullname + "." + name
    GpiObjHdl* new_obj = create_gpi_obj_from_handle(new_hdl, name, fq_name);
    if (new_obj == NULL) {
        vpi_free_object(new_hdl);
        LOG_ERROR("Unable to fetch object %s", fq_name.c_str());
        return NULL;
    }
    return new_obj;
```

Instead though, as then we use the fq_name as an internal string that we can use to compare objects equality and not as an implementation search which was the original intention.